File: src/service/GoodsGroupService.js
import service from './api';

class GoodsGroupService {

    search(search) {
        return service.post("/goods-groups/search", search);
    }

    get(id) {
        return service.get(`/goods-groups/${id}`);
    }

    create(data) {
        return service.post("/goods-groups", data);
    }

    update(id, data) {
        return service.put(`/goods-groups/${id}`, data);
    }

    delete(id) {
        return service.delete(`/goods-groups/${id}`);
    }

}

export default new GoodsGroupService();

File: back-end/vrp-preprocess/src/main/java/com/vrp/demo/service/GoodsGroupService.java
package com.vrp.demo.service;

import com.vrp.demo.entity.tenant.GoodsGroup;
import com.vrp.demo.exception.CustomException;
import com.vrp.demo.models.GoodsGroupModel;
import com.vrp.demo.models.search.GoodsGroupSearch;
import org.springframework.data.domain.Page;

import java.util.List;

public interface GoodsGroupService extends BaseService<GoodsGroup, Long> {

    public List<GoodsGroupModel> find(GoodsGroupSearch search);

    public Page<GoodsGroupModel> search(GoodsGroupSearch search);

    public GoodsGroupModel create(GoodsGroupModel goodsGroupModel) throws CustomException;

    public GoodsGroupModel update(GoodsGroupModel goodsGroupModel) throws CustomException;

    public int delete(Long id) throws CustomException;

    public GoodsGroupModel findOne(Long id);

    public List<GoodsGroup> getGoodsGroups(GoodsGroupSearch search);

}
